### ISSUE TITLE
ddcci-monitor-control@andr35: Multi-monitor support

### ISSUE THREAD

**Issue by tim-we** [linuxmint/cinnamon-spices-applets#3807]
@andr35

```
 * Applet version/Build date: 1.0.0 (2021-02-21 23:34:30)
 * Cinnamon version 4.8.6
 * Distribution: Linux Mint 20.1
 ```

It seems to me that the extension does not support multiple monitors. Could you add support for that? If not please let me know and I will try if I find time for that.

**Comment by Andr35** [comment]
You're right. Currently, the applet get and set the brightness and contrast of the default display supporting DDC/CI protocol.

It shouldn't be too difficult to adapt the applet. Internally it uses `ddcutil` so, to set the brightness/contrast values it should be sufficient to call `ddcutil setvcp` with the extra argument `-d X` with X the number of the display to configure.

Unfortunately, I don't have 2 external displays to test if any update would work.

**Comment by tim-we** [comment]
Thanks for replying. I just ran the `ddcutil` commands manually and that works as expected. However I am not sure what would be the best way of integrating this into the UI. Personally I would prefer just one brightness slider for each display and no default brightness buttons as those do not really make sense for multiple displays. So perhaps I should just create a new applet, specifically for the multi-monitor use case?

**Comment by Andr35** [comment]
The buttons are present because I developed the applet starting from my needs. The default buttons were useful to quickly switch from a brightness to another.

There are different solutions: 
- There could be only two default buttons on top of the applet that apply the brightness level to all the monitors.
- Or, the default buttons could be displayed only if they are enabled in applet's settings.

I wouldn't remove them. If you think that create a new applet works best, feel free to reuse snippets of code if you need it.

Probably, I'll implement one of the solutions but, currently, I haven't much time to spend on this.

**Comment by tim-we** [comment]
I wouldn't remove them either, there are probably users out there using those. One could keep them and have them set the brightness level for all monitors, like you suggested, but on different monitors different brightness levels result in different perceived brightness. I like the idea of having presets but IMO those would need a separate brightness level for each monitor. That complicates things in the UI and also I am not sure if the display number is guaranteed to always be the same. There is also the contrast slider which I haven't even considered yet.

The second solution is probably easier to implement but feels kinda weird to me as you have basically two different applets depending on the number of displays.

That is why I thought of creating a new applet, probably reusing some of your code. If I do so, I'll let you know.

**Comment by tim-we** [comment]
Hey I started working on a new applet. A small thing that I found that could also be improved in your applet is this:

https://github.com/linuxmint/cinnamon-spices-applets/blob/393d2da9b14ebbec55977122962e74b72a6b445d/ddcci-monitor-control%40andr35/files/ddcci-monitor-control%40andr35/applet.js#L40

You should change this to `this.set_applet_icon_symbolic_name("display-brightness");` to support light/dark mode.

Also I decided to use `--bus=$i` instead of `--display=$i`, it seems a bit faster to me (though I have not actually measured it tbh).

How does one publish an applet, just create a pull request on this repo?

**Comment by Andr35** [comment]
Yes, You create a pull request and wait for the approval of the maintainers of the repo.

https://github.com/linuxmint/cinnamon-spices-applets#additions
